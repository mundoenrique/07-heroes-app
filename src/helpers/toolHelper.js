import { NavLink } from 'react-router-dom';
import { heroes } from '../data/heroes';

export const NavBarLink = ({ children, to, className = '', ...props }) => {
	return (
		<>
			<NavLink
				className={({ isActive }) =>
					isActive ? `${className} active` : `${className}`
				}
				to={to}
				{...props}
			>
				{children}
			</NavLink>
		</>
	);
};

export const getHeroesById = (id) => {};

export const getHeroesByPublisher = (publisher) => {
	const validPublishers = ['DC Comics', 'Marvel Comics'];

	if (!validPublishers.includes(publisher)) {
		throw new Error(`${publisher} is not a valid publisher`);
	}

	return heroes.filter((hero) => hero.publisher === publisher);
};
