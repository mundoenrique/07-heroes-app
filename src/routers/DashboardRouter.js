import { Route, Routes } from 'react-router-dom';

import DcScreen from '../components/dc/DcScreen';
import HeroScreen from '../components/hero/HeroScreen';
import NavBar from '../components/iu/NavBar';
import MarvelScreen from '../components/marvel/MarvelScreen';
import SearchScreen from '../components/search/SearchScreen';

export default function DashboardRouter() {
	return (
		<>
			<NavBar />
			<Routes>
				<Route path="marvel" element={<MarvelScreen />} />
				<Route path="dc" element={<DcScreen />} />
				<Route path="search" element={<SearchScreen />} />
				<Route path="hero" element={<HeroScreen />} />
				<Route path="/" element={<MarvelScreen />} />
			</Routes>
		</>
	);
}
