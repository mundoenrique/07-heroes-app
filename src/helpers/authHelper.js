import { createContext } from 'react';

import { types } from './toolHelper';

export const AuthContext = createContext(null);

export const authReducer = (state = {}, action) => {
	const { login, logout } = types;
	const { type, payload } = action;

	switch (type) {
		case login:
			return {
				...payload,
				logged: true,
			};
		case logout:
			return {
				logged: false,
			};
		default:
			return state;
	}
};
