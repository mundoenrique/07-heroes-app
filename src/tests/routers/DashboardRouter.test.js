import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../helpers/authHelper';
import DashboardRouter from '../../routers/DashboardRouter';

describe('Pruebas sobre el <DashboardRouter />', () => {
	const contextValue = {
		user: {
			logged: true,
			name: 'Enrique',
		},
	};

	test('Debe mostrarse correctamente el <MarvelScrren />', () => {
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<MemoryRouter initialEntries={['/']}>
					<DashboardRouter />
				</MemoryRouter>
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('h1').text().trim()).toBe('MarvelScreen');
	});
	test('Debe mostrarse correctamente el <Dc />', () => {
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<MemoryRouter initialEntries={['/dc']}>
					<DashboardRouter />
				</MemoryRouter>
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('h1').text().trim()).toBe('DcScreen');
	});
});
