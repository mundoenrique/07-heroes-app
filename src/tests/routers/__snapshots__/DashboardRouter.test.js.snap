// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <DashboardRouter /> Debe mostrarse correctamente el <Dc /> 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Enrique
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
        DcScreen
      </h1>,
      Array [
        <hr />,
        <div
          className="row row-cols-1 row-cols-md-3 g-4 animate__animated animate__fadeIn"
        >
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Batman"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-batman.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Batman
                    </h5>
                    <p
                      className="card-text"
                    >
                      Bruce Wayne
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Detective Comics #27
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-batman"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Superman"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-superman.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Superman
                    </h5>
                    <p
                      className="card-text"
                    >
                      Kal-El
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Action Comics #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-superman"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Flash"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-flash.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Flash
                    </h5>
                    <p
                      className="card-text"
                    >
                      Jay Garrick
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Flash Comics #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-flash"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Green Lantern"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-green.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Green Lantern
                    </h5>
                    <p
                      className="card-text"
                    >
                      Alan Scott
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        All-American Comics #16
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-green"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Green Arrow"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-arrow.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Green Arrow
                    </h5>
                    <p
                      className="card-text"
                    >
                      Oliver Queen
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        More Fun Comics #73
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-arrow"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Wonder Woman"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-wonder.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Wonder Woman
                    </h5>
                    <p
                      className="card-text"
                    >
                      Princess Diana
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        All Star Comics #8
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-wonder"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Martian Manhunter"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-martian.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Martian Manhunter
                    </h5>
                    <p
                      className="card-text"
                    >
                      J'onn J'onzz
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Detective Comics #225
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-martian"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Robin/Nightwing"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-robin.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Robin/Nightwing
                    </h5>
                    <p
                      className="card-text"
                    >
                      Dick Grayson
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Detective Comics #38
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-robin"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Blue Beetle"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-blue.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Blue Beetle
                    </h5>
                    <p
                      className="card-text"
                    >
                      Dan Garret
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Mystery Men Comics #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-blue"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Black Canary"
                    className="img-fluid rounded-start"
                    src="/assets/images/dc-black.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Black Canary
                    </h5>
                    <p
                      className="card-text"
                    >
                      Dinah Drake
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Flash Comics #86
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/dc-black"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>,
      ],
    ]
  </div>,
]
`;

exports[`Pruebas sobre el <DashboardRouter /> Debe mostrarse correctamente el <MarvelScrren /> 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Enrique
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
        MarvelScreen
      </h1>,
      Array [
        <hr />,
        <div
          className="row row-cols-1 row-cols-md-3 g-4 animate__animated animate__fadeIn"
        >
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Spider Man"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Spider Man
                    </h5>
                    <p
                      className="card-text"
                    >
                      Peter Parker
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Amazing Fantasy #15
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-spider"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Captain America"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Captain America
                    </h5>
                    <p
                      className="card-text"
                    >
                      Steve Rogers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Captain America Comics #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-captain"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Iron Man"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Iron Man
                    </h5>
                    <p
                      className="card-text"
                    >
                      Tony Stark
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #39
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-iron"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Thor"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Thor
                    </h5>
                    <p
                      className="card-text"
                    >
                      Thor Odinson
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Journey into Myster #83
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-thor"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Hulk"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hulk
                    </h5>
                    <p
                      className="card-text"
                    >
                      Bruce Banner
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-hulk"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Wolverine"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Wolverine
                    </h5>
                    <p
                      className="card-text"
                    >
                      James Howlett
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #180
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-wolverine"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Daredevil"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Daredevil
                    </h5>
                    <p
                      className="card-text"
                    >
                      Matthew Michael Murdock
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Daredevil #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-daredevil"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Hawkeye"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hawkeye
                    </h5>
                    <p
                      className="card-text"
                    >
                      Clinton Francis Barton
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #57
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-hawkeye"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Cyclops"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Cyclops
                    </h5>
                    <p
                      className="card-text"
                    >
                      Scott Summers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        X-Men #1
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-cyclops"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="card mb-3"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Silver Surfer"
                    className="img-fluid rounded-start"
                    src="/assets/images/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Silver Surfer
                    </h5>
                    <p
                      className="card-text"
                    >
                      Norrin Radd
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Fantastic Four #48
                      </small>
                    </p>
                    <a
                      className="btn btn-link"
                      href="/hero/marvel-silver"
                      onClick={[Function]}
                    >
                      Mas...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>,
      ],
    ]
  </div>,
]
`;
